commons-rest-test/src/main/java/com/mule/connectors/commons/rest/test/provider/TestCase.java

```diff
@@ -22,18 +22,31 @@
 public class TestCase {
 
     private final Request request;
-    private final List<Matcher<? super RequestAndResponse>> responseAssertions;
+    private final List<RequestAndResponseAssertion> assertions;
+    private final List<Request> setupRequestList;
+    private final List<Request> tearDownRequestList;
 
     @JsonCreator
-    public TestCase(@JsonProperty(value = "request", required = true) Request request,
-            @JsonProperty(value = "assertions", required = true) List<RequestAndResponseAssertion> responseAssertionsParam) {
+    public TestCase(@JsonProperty(value = "before", required = true) List<Request> setupRequestList,
+            @JsonProperty(value = "request", required = true) Request request,
+            @JsonProperty(value = "after", required = true) List<Request> tearDownRequestList,
+            @JsonProperty(value = "assertions", required = true) List<RequestAndResponseAssertion> assertions) {
+        this.setupRequestList = Optional.fromNullable(tearDownRequestList).or(new ArrayList<Request>());
         this.request = request;
-        List<Matcher<? super RequestAndResponse>> assertions = new ArrayList<>();
-        assertions.addAll(Optional.fromNullable(responseAssertionsParam).or(new ArrayList<RequestAndResponseAssertion>()));
-        this.responseAssertions = assertions;
+        this.tearDownRequestList = Optional.fromNullable(setupRequestList).or(new ArrayList<Request>());
```

Comment:
How about using emptyList() instead of newArrayList(), here and in the next line?